crates/bevy_math/src/curve/mod.rs

```diff
@@ -1015,6 +1305,51 @@ mod tests {
         assert_eq!(mapped_curve.domain(), interval(0.0, 1.0).unwrap());
     }
 
+    #[test]
+    fn reverse() {
+        let curve = function_curve(Interval::new(0.0, 1.0).unwrap(), |t| t * 3.0 + 1.0);
+        let rev_curve = curve.reverse().unwrap();
+        assert_eq!(rev_curve.sample_unchecked(0.0), 1.0 * 3.0 + 1.0);
+        assert_eq!(rev_curve.sample_unchecked(0.5), 0.5 * 3.0 + 1.0);
+        assert_eq!(rev_curve.sample_unchecked(1.0), 0.0 * 3.0 + 1.0);
+    }
+
+    #[test]
+    fn repeat() {
+        let curve = function_curve(Interval::new(0.0, 1.0).unwrap(), |t| t * 3.0 + 1.0);
+        let repeat_curve = curve.repeat(RepeatMode::Fixed(1)).unwrap();
+        assert_eq!(repeat_curve.sample_unchecked(0.0), 0.0 * 3.0 + 1.0);
+        assert_eq!(repeat_curve.sample_unchecked(0.5), 0.5 * 3.0 + 1.0);
+        assert_eq!(repeat_curve.sample_unchecked(0.99), 0.99 * 3.0 + 1.0);
+        assert_eq!(repeat_curve.sample_unchecked(1.0), 0.0 * 3.0 + 1.0);
+        assert_eq!(repeat_curve.sample_unchecked(1.5), 0.5 * 3.0 + 1.0);
+        assert_eq!(repeat_curve.sample_unchecked(1.99), 0.99 * 3.0 + 1.0);
+        assert_eq!(repeat_curve.sample_unchecked(2.0), 0.0 * 3.0 + 1.0);
```

Comment:
I find it confusing that the final value of the `RepeatCuve` is the initial value of the base curve. I would expect it to be the final value of the base curve, in this case I would expect 

```rust
assert_eq!(repeat_curve.sample_unchecked(2.0), 1.0 * 3.0 + 1.0);
```